python/cudf/cudf/_lib/pylibcudf/copying.pyx

```diff
@@ -55,3 +66,198 @@ cpdef Table gather(
             )
         )
     return Table.from_libcudf(move(c_result))
+
+
+cpdef Column shift(Column input, size_type offset, Scalar fill_values):
+    cdef unique_ptr[column] c_result
+    with nogil:
+        c_result = move(
+            cpp_copying.shift(
+                input.view(),
+                offset,
+                dereference(fill_values.c_obj)
+            )
+        )
+    return Column.from_libcudf(move(c_result))
+
+
+cpdef Table scatter(object source, Column scatter_map, Table target_table):
+    cdef unique_ptr[table] c_result
+    cdef vector[reference_wrapper[constscalar]] source_scalars
+    cdef Scalar slr
+
+    if isinstance(source, Table):
+        with nogil:
+            c_result = move(
+                cpp_copying.scatter(
+                    (<Table> source).view(),
+                    scatter_map.view(),
+                    target_table.view(),
+                )
+            )
+    elif isinstance(source, list):  # TODO: is list too restrictive?
+        for slr in source:
+            source_scalars.push_back(
+                reference_wrapper[constscalar](dereference(slr.c_obj))
+            )
```

Comment:
**suggestion** This pattern already repeats twice, let's pull it into a utility function.

Reply:
Done.